clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeliverArraysExecutor.scala

```diff
@@ -22,24 +22,57 @@ class DeliverArraysExecutor(deliverCommand: DeliverArrays)(implicit ec: Executio
     extends DeliverExecutor(deliverCommand) {
 
   override protected[dispatch] def buildDelivery(
-    deliveredMetadata: ArraysMetadata,
-    moveOps: immutable.Seq[IoOp]
+    deliveredMetadata: ArraysMetadata
   ): Source[(ArraysMetadata, immutable.Seq[IoOp]), NotUsed] = {
-    (deliveredMetadata.grnIdatPath, deliveredMetadata.redIdatPath) match {
-      case (Some(grn), Some(red)) => {
+    (
+      deliveredMetadata.vcfPath,
+      deliveredMetadata.vcfIndexPath,
+      deliveredMetadata.gtcPath,
+      deliveredMetadata.grnIdatPath,
+      deliveredMetadata.redIdatPath
+    ) match {
+      case (Some(vcf), Some(index), Some(gtc), Some(grn), Some(red)) => {
+        val vcfMove = Metadata.buildFilePath(
+          vcf,
+          deliverCommand.destination,
+          ArraysExtensions.VcfGzExtension,
+          moveCommand.newBasename
+        )
+        val vcfIndexMove = Metadata.buildFilePath(
+          index,
+          deliverCommand.destination,
+          ArraysExtensions.VcfGzTbiExtension,
+          moveCommand.newBasename
+        )
+        val gtcMove = Metadata.buildFilePath(
+          gtc,
+          deliverCommand.destination,
+          ArraysExtensions.GtcExtension,
+          moveCommand.newBasename
+        )
+        val moves = immutable
+          .Seq(
+            MoveOp(vcf, vcfMove),
+            MoveOp(index, vcfIndexMove),
+            MoveOp(gtc, gtcMove)
+          )
+          .filterNot { op =>
+            op.src.equals(op.dest)
```

Comment:
This filter would be worth defining once in the `MoveExecutor`, in a separate follow-up method. We'd need to make it handle all 3 cases (but that should be OK). Both moves and copies get dropped if src == dest, and writes always happen(?).

Reply:
check.